feature 283 - Search bar place holder text now displays hint text

brianandres2 opened the pull request:
The place holder text in the search bar now displays hint text, however I ran into a few problems

Any html formatting done on the hint text (i.e. bolding) showed up in the text as'< b> < /b>, and as far as I could tell there was no easy solution to make the html tag do what we want in the place holder text. To fix this, I added a new variable to 'fields' called 'help_text_unformatted' where I removed all html tags and edited the help text to make it shorter if it was too long for the place holder text. 

The only place where the text being too long was an issue was with Author and Adoption Level. With Author, I just removed the example part of the help text (the 'e.g.' part), but with *Adoption Level I had to get rid of part of the explanation of the different levels. Let me know if you want me to do anything different for that one.

Additionally, I also changed part of the Title text in both the formatted and unformatted help text to make the grammar a little better, let me know if you want me to revert it:

Original:    'Enter if possible at least the first five words of the title (try all title if you have it)'
Changed:  'Enter if possible at least the first five words of the title (try the entire title if you have it)'
(changed 'all title' to 'the entire title')

How the place holder text looks:

<img width="1680" alt="Screen Shot 2022-08-09 at 11 50 06 AM" src="https://user-images.githubusercontent.com/107065749/183699055-0b04811a-13aa-4e0a-8f0d-346d197a90f8.png">

*With Adoption Level, I changed it from:

'Search for the following Adoption terms: Novel (no adoption outside originators); Validated (tested by third parties); Organizational; Multi-Organizational; National; International'

to: 'Search Adoption Terms: Novel; Validated; Organizational; Multi-Organizational; National; International' 

let me know if there is anything different you want me to do with Adoption Level, or if you want me to go through doing the place holder text a different way, I am more than happy to try a different approach

It should also be noted that if the screen is smaller some of the place holder text will get cut off

brianandres2 committed:
search place holder text now displays hint text

paulpilone reviewed (commented):
@brianandres2 How hard would it be to add media breakpoints to shrink the hit font size if the screen gets small - or another option would be to add an ellipse if the text overflows?

Not a huge deal since it depends on screen size but thinking of ways to present it _slightly_ better on smaller screens.

Otherwise this looks good.

brianandres2 committed:
Placeholder text now shrinks and has ellipsis

brianandres2 commented:
@paulpilone I added media breakpoints so the font-size shrinks as the screen gets smaller, and also added ellipsis when the width becomes too small (and where the font size would become too small). This works well for all of them, though 'Adoption Level' still cuts out a little earlier than all the other fields.

However, on the landing page, 3 of the fields help text are too big since the search bar is smaller. For the landing page should I make the font size a lot smaller for the placeholder text, or leave it as it is where the text cuts out with ellipsis for some of the placeholder text, or just make the placeholder text "Search OceanBestPractices" for all the fields (on landing page only)?

paulpilone reviewed (approved):
This looks good with these changes. The hint text is just too long for some items and I think this is as good as it'll get.
